client/src/home/edit_form/EditForm.js

```diff
@@ -0,0 +1,201 @@
+    const [lbPerHouse, setLbPerHouse] = useState(0);
+    const [weight, setWeight] = useState(0);
+    const [outByDate, setOutByDate] = useState(new Date());
+    
+    const [cookies] = useCookies(["token"]);
+    const [showWeightWarning, setWeightWarning] = useState(false);
+    const [showitemNameWarning, setitemNameWarning] = useState(false);
+    const [showbatchIdWarning, setbatchIdWarning] = useState(false);
+    // Logic for adding recommendations will need to be added
+    const [rec, setRec] = useState([]);
+    const [itemList, setItemList] = useState([]);
+  
+    useEffect(() => {
+      setItemList(props.itemList);
+    }, [props.itemList])
+  
+    return (
+      <div className="edit_form_container">
+        <Form className="edit_form">
+          <Form.Group className="input_field_container">
+            <Form.Group>
+              <Form.Label> Item Name</Form.Label>
+              <Form.Control
+                
+                
+                onChange={event => {
+                  let string = event.target.value.trim().toLowerCase();
+                  if(string.length === 0){
+                    setitemNameWarning(true);
+                  }
+                  else{ 
+                    setitemNameWarning(false);
+                  }
+                  setItemName(string);
+                }}
+                onBlur={e => {
+                    if(itemName.length === 0){
+                      setitemNameWarning(true);
+                    }
+                    else{
+                      setitemNameWarning(false);
+                    }
+                  }}
+                
+                placeholder="e.g. Apples"
+                autoFocus
+                styling={{
+                  border: "1px solid #cbcdd1",
+                  borderRadius: '5px',
+                  boxShadow: "0 0 0 0.2rem rgba(0,123,255,.25)",
+                  hoverBackgroundColor: "#80bdff",
+                  height: '40px',
+                }}
+              />
+              {showitemNameWarning && (
+                <Form.Text className="warning">Item name is required</Form.Text>
+              )}
+
+            </Form.Group>
+  
+            <Form.Group>
+              <Form.Label> Batch ID</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="e.g. 1234"
+                
+                onBlur={(e) => {
+                  if(newBatchId.length === 4){
+                    setbatchIdWarning(false);
+                  }
+                  else{
+                    setbatchIdWarning(true);
+                  }
+                }}
+                onChange={(event) => {
+                  let val = event.target.value.trim().toLowerCase();
+                  if(val.length === 4){
+                    setbatchIdWarning(false);
+                  }
+                  else{
+                    setbatchIdWarning(true);
+                  }
+                  setOldBatchId(val);
```

Comment:
I just want to confirm that you are setting oldBatchId because the calendar isn't set up yet? In the future you will only setNewBatchId, and oldBatchId will stay as whatever the batchId was?

Reply:
Yeah, that's what we were thinking👍